core/src/impl/Kokkos_Core.cpp

```diff
@@ -799,6 +806,74 @@ void parse_environment_variables(InitArguments& arguments) {
 
 }  // namespace
 
+int get_ctest_gpu(const char* local_rank_str)
```

Comment:
Please document what this function does.

Also what is the rational for returning zero if there is any problem with the resource specification rather than throwing an exception?

Reply:
I can make it throw an exception instead.